### DISCUSSION TITLE
Add Temperature & Humidity State Colors

### DISCUSSION THREAD

**Discussion by Nixon506E** [home-assistant/frontend#22140] [Other feature requests]
I have been using temperature and humidity chips and heading entities and would love to be able to customize their icons with colors representing their state. 

We could use the AQI color ranges provided here: https://www.airnow.gov/aqi/aqi-basics/

Green: 0-50
Yellow: 51-100
Orange: 101-150
Red: 151-200
Purple: 201-300
Maroon: 301+

We could use the temperature color ranges provided here:
<img width="422" alt="Color Ranges" src="https://github.com/user-attachments/assets/00876f59-9039-4125-a956-539c9a430ffc">

**Comment by ildar170975** [comment]
1.  This colorization can be achieved by card-mod or customUI. 
2. The color scheme for a temperature which you provided is for AIR. But the device_class temperature may be used for many other applications where this scheme may be not suitable. Similarly - for a humidity: in a hotbed there are different requirements for a humidity , thus you would need a different color scheme.

**Reply by Nixon506E** [reply]
1. Trying to eliminate the need for custom modifications to the UI is always preferred
2. I could being able to provide custom colors in entity customization but you must start somewhere

Maybe the preference would be introducing the ability to set an icon_color directly on the entity and be able to customize it or set it via a template

**Reply by ildar170975** [reply]
> would be introducing the ability to set an icon_color directly on the entity and be able to customize it or set it via a template

could be a useful universal addition.
